tests/test_generics.py

```diff
@@ -2299,3 +2301,65 @@ class Model(BaseModel, Generic[T]):
         'title': 'Model[test_generic_intenum_bound.<locals>.MyEnum]',
         'type': 'object',
     }
+
+
+@pytest.mark.skipif(sys.version_info < (3, 11), reason='requires python 3.11 or higher')
+@pytest.mark.xfail(reason='TODO: Variadic generic parametrization is not supported yet')
+def test_variadic_generic_init():
+    class ComponentModel(BaseModel):
+        pass
+
+    class Wrench(ComponentModel):
+        pass
+
+    class Screwdriver(ComponentModel):
+        pass
+
+    ComponentVar = TypeVar('ComponentVar', bound=ComponentModel)
+    NumberOfComponents = TypeVarTuple('NumberOfComponents')
+
+    class VariadicToolbox(BaseModel, Generic[ComponentVar, *NumberOfComponents]):
+        main_component: ComponentVar
+        left_component_pocket: Optional[list[ComponentVar, ...]] = Field(default_factory=list)
+        right_component_pocket: Optional[list[ComponentVar, ...]] = Field(default_factory=list)
+
+        @computed_field
+        @property
+        def all_components(self) -> list[ComponentVar, *NumberOfComponents]:
```

Comment:
`list` isn't a variadic generic is it, even in 3.11? I think probably you want `tuple` here? Or maybe I missed something else that changed? (Same for the things above which have `list[ComponentVar, ...]`)

Reply:
My mistake, list a mutable structure so variadics don't make sense here.